### ISSUE TITLE
Tabs tray remains improperly displayed at all times

### ISSUE THREAD

**Issue by delia-pop** [mozilla-mobile/focus-android#6729]
## Steps to reproduce
1. Have at least 2 tabs opened in Focus.
2. Tap the tab counter.
3. Send Focus in background.
4. Bring Focus to foreground and observe that the tab counter is still displayed. 
5. From Gmail, open a link in Focus and observe the tab counter.

### Expected behavior
After Step 5, when a new tab is opened in Focus, the tabs counter should not be already invoked, or at least should be updated with the new tab added.
### Actual behavior
After Step 5, when a new tab is opened in Focus, the tab counter is improperly displayed, only showing the previous tabs, confusing the user.

### Device information

* Android device: Xiaomi Redmi Note 8T (Android 9).
* Focus version: Focus Nightly 100.0a1 from 03/23, Focus Beta 99.0.0-beta.4, RC 98.3.0.

https://user-images.githubusercontent.com/89388888/159693568-823ab0e3-8721-4b96-8cf5-950f9e542a76.mp4

**Comment by delia-pop** [comment]
Please note that this issue can be reproduce with multiple floating context menus invoked throughout the application when a tab is opened in Focus from an external app.

![Untitled](https://user-images.githubusercontent.com/89388888/159718441-908a823c-d349-4b55-9e9c-9ce4600aaffc.png)

**Comment by lobontiumira** [comment]
Verified on the latest Focus Nightly from 3/30 with multiple devices:
* Lenovo Tab M10 (Android 10),
* Samsung Galaxy Note 8 (Android 9),
* Google Pixel (Android 10).

The tabs tray is now dismissed after putting the app in the background and reopen it.
The tabs tray is dismissed when opening a tab from a different app.

However, the ETP panel is not dismissed when opening a tab from an external app, like Gmail:

![Screenshot_20220330-140857_Firefox Focus Nightly](https://user-images.githubusercontent.com/40694158/160821941-3cd12158-f1bf-4167-b690-531207969194.jpg)

@ionutbedregeanu please review. LEt me know if I should file a new issue regarding the ETP panel still displayed.

**Comment by ionutbedregeanu** [comment]
@lobontiumira Please create a separate ticket for ETP. Thanks! 🙏
